src/fs_utf8/dir.rs

```diff
@@ -419,6 +455,52 @@ impl Dir {
             cap_std: self.cap_std.try_clone()?,
         })
     }
+
+    /// Returns `true` if the path points at an existing entity.
+    ///
+    /// This corresponds to [`std::path::Path::exists`], but only
+    /// accesses paths relative to `self`.
+    ///
+    /// [`std::path::Path::exists`]: https://doc.rust-lang.org/std/path/struct.Path.html#method.exists
+    #[inline]
+    pub fn exists<P: AsRef<str>>(&self, path: P) -> bool {
+        let path = match from_utf8(&path) {
+            Ok(path) => path,
+            Err(_) => return false,
```

Comment:
I think it might useful explaining in the docs that if the `path` is not a valid UTF8, this will return `false`.

Reply:
There is a subtlety here: `path` is an `AsRef<str>`, so it's already guaranteed to be valid UTF-8. `fs_utf8` is using arf-strings to encode non-UTF-8 paths as valid UTF-8 strings. So it isn't "there might be a file there, but we're hiding it from you". You can always name and access that file, you just need to use the proper encoding (and that's what you're opting into by using the `fs_utf8` module :-)). `from_utf8` only fails if the arf-string encoding is invalid.

Reply:
That's fair enough. However, I'd still vote in favour of leaving a doc comment to remind the user about the possible cause of the failure. Having said that, perhaps this should be documented in the main `fs_utf8` crate/module's manifest instead of above every method? I know this should be obvious and this is what the user implicitly signs up for when using `fs_utf8`, but I've seen it countless of times when a tiny detail like this would go unnoticed just to rediscover this basic truth after hours of debugging.

Reply:
Good idea. I've now added a paragraph to the module comment expalining what the `fs_utf8` module is all about.

Reply:
Nice! This is perfect, thanks!